engine/base/kafka/src/main/scala/pl/touk/nussknacker/engine/baseengine/kafka/NuKafkaEngineApp.scala

```diff
@@ -0,0 +1,80 @@
+package pl.touk.nussknacker.engine.baseengine.kafka
+
+import com.typesafe.config.{Config, ConfigFactory}
+import org.apache.commons.io.FileUtils
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.api.deployment.DeploymentData
+import pl.touk.nussknacker.engine.api.{JobData, ProcessVersion}
+import pl.touk.nussknacker.engine.baseengine.api.runtimecontext.EngineRuntimeContextPreparer
+import pl.touk.nussknacker.engine.baseengine.metrics.NoOpMetricsProvider
+import pl.touk.nussknacker.engine.canonize.ProcessCanonizer
+import pl.touk.nussknacker.engine.marshall.ProcessMarshaller
+import pl.touk.nussknacker.engine.util.config.CustomFicusInstances._
+import pl.touk.nussknacker.engine.util.loader.ModelClassLoader
+
+import java.io.File
+import java.lang.Thread.UncaughtExceptionHandler
+import java.net.URL
+import java.nio.file.Path
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.Using
+
+object NuKafkaEngineApp extends App {
+
+  val scenarioFileLocation = parseArgs
+
+  val scenario = parseScenario
+
+  val scenarioInterpreter = prepareScenarioInterpreter
+
+  Using.resource(scenarioInterpreter) { interpreter =>
+    interpreter.run()
+  }
+
+  private def parseArgs = {
+    if (args.length < 1) {
+      System.err.println("Missing scenario_file_location argument!")
+      System.err.println("")
+      System.err.println("Usage: ./run.sh scenario_file_location.json")
+      sys.exit(1)
+    }
+
+    Path.of(args(0))
+  }
+
+  private def parseScenario = {
+    val validatedCanonicalScenario = ProcessMarshaller.fromJson(FileUtils.readFileToString(scenarioFileLocation.toFile))
+
+    val canonicalScenario = validatedCanonicalScenario.valueOr { err =>
+      System.err.println("Scenario file is not a valid json")
+      System.err.println(s"Errors found: ${err.msg}")
+      sys.exit(2)
+    }
+    val validatedScenario = ProcessCanonizer.uncanonize(canonicalScenario)
+    validatedScenario.valueOr { err =>
+      System.err.println(s"Scenario uncanonization error: ${err.toList.mkString(", ")}")
+      sys.exit(3)
+    }
+  }
+
+  private def prepareScenarioInterpreter = {
+    // TODO Should it be extracted from designer?
+    val engineConfig = ConfigFactory.load(ConfigFactory.parseFile(new File(System.getProperty("nussknacker.config.locations"))))
+    val modelConfig: Config = engineConfig.getConfig("modelConfig")
+
+    val modelData = ModelData(modelConfig, ModelClassLoader(modelConfig.as[List[URL]]("classPath")))
+    // TODO Use correct MetricProvider
+    val preparer = new EngineRuntimeContextPreparer(NoOpMetricsProvider)
+    // TODO Pass correct ProcessVersion and DeploymentData
+    val jobData = JobData(scenario.metaData, ProcessVersion.empty, DeploymentData.empty)
+
+    val exceptionHandler = new UncaughtExceptionHandler {
+      override def uncaughtException(t: Thread, e: Throwable): Unit = {
+        // TODO Is it ok?
+        throw e
```

Comment:
Can we add some error to log?

Reply:
added

Reply:
Also am I right that we should do `sys.exit` after logging? I've found your note about that in test - added